### ISSUE TITLE
print(list(cache.keys()))

### ISSUE THREAD

**Issue by Mfun82** [luo3300612/Visualizer#3]
您好！请问为什么我print(list(cache.keys()))的输出结果是一个[]，没有任何东西，这会是什么情况导致的呢？打扰了，感谢！

**Comment by Mfun82** [comment]
get_local.clear()
    with torch.no_grad():
        feat = model(img)
    # 获取attention map
    cache = get_local.cache
    print(list(cache.keys()))

这是我的部分代码，谢谢！

**Comment by luo3300612** [comment]
模型装饰的代码是怎样的呢

**Comment by Mfun82** [comment]
> 模型装饰的代码是怎样的呢

class Attention(nn.Module):
    def __init__(self, dim, num_heads=8, qkv_bias=False, qk_scale=None, attn_drop=0., proj_drop=0.):
        super().__init__()
        self.num_heads = num_heads
        head_dim = dim // num_heads
        # NOTE scale factor was wrong in my original version, can set manually to be compat with prev weights
        self.scale = qk_scale or head_dim ** -0.5

        self.qkv = nn.Linear(dim, dim * 3, bias=qkv_bias)
        self.attn_drop = nn.Dropout(attn_drop)
        self.proj = nn.Linear(dim, dim)
        self.proj_drop = nn.Dropout(proj_drop)

    @get_local('attn')
    def forward(self, x):
        B, N, C = x.shape
        qkv = self.qkv(x).reshape(B, N, 3, self.num_heads, C // self.num_heads).permute(2, 0, 3, 1, 4)
        q, k, v = qkv[0], qkv[1], qkv[2]   # make torchscript happy (cannot use tensor as tuple)

        attn = (q @ k.transpose(-2, -1)) * self.scale
        attn = attn.softmax(dim=-1)
        attn = self.attn_drop(attn)

        x = (attn @ v).transpose(1, 2).reshape(B, N, C)
        x = self.proj(x)
        x = self.proj_drop(x)
        return x

以上！感谢！

**Comment by luo3300612** [comment]
你有在import model之前先调用`get_local.activate()`吗

**Comment by Mfun82** [comment]
> 你有在import model之前先调用`get_local.activate()`吗

是的，我在最上方激活了

**Comment by luo3300612** [comment]
奇怪那按理说不应该会这样，我的notebook你跑了会这样吗，你确定被装饰的函数被你import的模型调用了吗，里面加一行print什么的试试看会不会有输出

**Comment by Mfun82** [comment]
> 奇怪那按理说不应该会这样，我的notebook你跑了会这样吗，你确定被装饰的函数被你import的模型调用了吗，里面加一行print什么的试试看会不会有输出

发现问题了！是因为我在另外一个文件里面也导入了模型，在两个文件里都激活一下就可以了！谢谢您的时间！

**Comment by jiangke-sr** [comment]
我在notebook上也出现了，重新启动然后运行就ok
